db/migration/src/m20240301_000014_create_ticket_stats_with_channel_id.rs

```diff
@@ -0,0 +1,123 @@
+                            .not_null()
+                            .unique_key(),
+                    )
+                    .col(ColumnDef::new(TicketStatistics::LastUpdated).timestamp().null())
+                    .col(
+                        ColumnDef::new(TicketStatistics::LosingTickets)
+                            .not_null()
+                            .integer()
+                            .default(0),
+                    )
+                    .col(
+                        ColumnDef::new(TicketStatistics::RedeemedTickets)
+                            .not_null()
+                            .integer()
+                            .default(0),
+                    )
+                    .col(
+                        ColumnDef::new(TicketStatistics::RedeemedValue)
+                            .binary_len(12)
+                            .not_null()
+                            .default(vec![0u8; 12]),
+                    )
+                    .col(
+                        ColumnDef::new(TicketStatistics::NeglectedTickets)
+                            .not_null()
+                            .integer()
+                            .default(0),
+                    )
+                    .col(
+                        ColumnDef::new(TicketStatistics::NeglectedValue)
+                            .binary_len(12)
+                            .not_null()
+                            .default(vec![0u8; 12]),
+                    )
+                    .col(
+                        ColumnDef::new(TicketStatistics::RejectedTickets)
+                            .not_null()
+                            .integer()
+                            .default(0),
+                    )
+                    .col(
+                        ColumnDef::new(TicketStatistics::RejectedValue)
+                            .binary_len(12)
+                            .not_null()
+                            .default(vec![0u8; 12]),
+                    )
+                    .to_owned(),
+            )
+            .await?;
+
+        let conn = manager.get_connection();
+
+        conn.execute_unprepared(
+            r#"
+            CREATE TRIGGER IF NOT EXISTS trig_ticket_stats_update_timestamp
+                AFTER UPDATE
+                ON ticket_statistics
+                FOR EACH ROW
+                WHEN OLD.last_updated IS NULL OR NEW.last_updated < OLD.last_updated --- avoid infinite loop
+            BEGIN
+                UPDATE ticket_statistics SET last_updated=CURRENT_TIMESTAMP WHERE id=OLD.id;
+            END;
+        "#,
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        let conn = manager.get_connection();
+
+        conn.execute_unprepared("DROP TRIGGER trig_ticket_stats_update_timestamp;")
+            .await?;
+
+        manager
+            .drop_table(Table::drop().table(TicketStatistics::Table).to_owned())
+            .await
+    }
+}
```

Comment:
This will not result in the same DB though. The previous version of the database needs to be reconstructed. Also, do we not need to migrate the data?

Reply:
Yes we sacrifice the data in the stats table.